packages/venia-ui/lib/components/OrderHistoryPage/OrderDetails/item.js

```diff
@@ -0,0 +1,70 @@
+import React, { useMemo } from 'react';
+
+import { Link, resourceUrl } from '@magento/venia-drivers';
+import { mergeClasses } from '@magento/venia-ui/lib/classify';
+
+import Button from '../../Button';
+import ProductOptions from '../../LegacyMiniCart/productOptions';
+import Image from '../../Image';
+
+import defaultClasses from './item.css';
+
+const Item = props => {
+    const {
+        product_name,
+        product_sale_price,
+        quantity_ordered,
+        selected_options,
+        thumbnail,
+        url_key,
+        url_suffix
+    } = props;
+    const { url } = thumbnail;
+    const itemLink = useMemo(() => resourceUrl(`/${url_key}${url_suffix}`), [
+        url_key,
+        url_suffix
+    ]);
+    const mappedOptions = useMemo(
+        () =>
+            selected_options.map(option => ({
+                option_label: option.label,
+                value_label: option.value
+            })),
+        [selected_options]
+    );
+    const classes = mergeClasses(defaultClasses, props.classes);
+
+    return (
+        <div className={classes.root}>
+            <Link className={classes.thumbnailContainer} to={itemLink}>
+                <Image
+                    alt={product_name}
+                    classes={{ root: classes.thumbnail }}
+                    width={50}
+                    resource={url}
+                />
+            </Link>
+            <Link className={classes.name} to={itemLink}>
+                {product_name}
+            </Link>
+            <ProductOptions
+                options={mappedOptions}
+                classes={{
+                    options: classes.options
+                }}
+            />
+            <span
+                className={classes.quantity}
+            >{`Qty : ${quantity_ordered}`}</span>
+            <span className={classes.price}>{product_sale_price}</span>
+            <Button onClick={() => {}} className={classes.buyAgainButton}>
+                {'Buy Again'}
+            </Button>
+            <Button onClick={() => {}} className={classes.returnThisButton}>
+                {'Return This'}
```

Comment:
Per conversation in dev time, lets try returning `strings` from the talons for all of these components.

Something like:

```
strings: {
  returnThisButtonText: formatMessage({ id: orderHistoryPage.orderDetails.item.returnThisButtonText })
  ...
}
```

This does raise a point of how to properly identify the strings. If we just use `component.thing` we will most likely have overlap.

Reply:
What if a component does not have a talon but has constant strings? For instance, [shippingInformation.js](https://github.com/magento/pwa-studio/blob/revanth/order-history-expanded/packages/venia-ui/lib/components/OrderHistoryPage/OrderDetails/shippingInformation.js#L23) does not have a talon but has a heading called `Shipping Information`.

Reply:
There was discussion in chat yesterday about not doing this - and your point is valid. I think we need to rediscuss in dev time, if at all. I lean towards just doing formatting in the view.

Reply:
Me too, let discuss about this today 👍 .